---
title: "QC and evaluation of long read data"
teaching: 30
exercises: 60-90
questions:
- "How to determine basic statistics on read sets"
- "What do these statistics mean: are they good or bad?"
- "What are read technology specific errors?"  
objectives:
- "Learn how to get and interpret read set statistics"
- "Evaluate read technology specific errors based on read mappings"
keypoints:
- "PacBio and Nanopore have read lengths of many kilobases"
- "In practice quality of input DNA, library prep. and many other factors will determine read length distribution"
- "PacBio and Nanopore have different types of errors"
apps:
- "./tools/assembly-stats-master/build/assembly-stats" 
---

To run the quality control on the read sets we use assembly-stats:
~~~
gt-workshop@gtworkshop-VirtualBox:~$ ./tools/assembly-stats-master/build/assembly-stats 
usage: stats [options] <list of fasta/q files>

Reports sequence length statistics from fasta and/or fastq files

options:
-l <int>
	Minimum length cutoff for each sequence.
	Sequences shorter than the cutoff will be ignored [1]
-s
	Print 'grep friendly' output
-t
	Print tab-delimited output
-u
	Print tab-delimited output with no header line
-v
	Print version and exit
~~~
{: .bash}

> ## Read statistics
> The PacBio reads are in this file: data/raw_data/pacbio_reads.fasta.
> Get the read statistics on this data set and try to interpet the results
> > ## Solution
> > ~~~
> > gt-workshop@gtworkshop-VirtualBox:~$ ./tools/assembly-stats-master/build/assembly-stats data/raw_data/pacbio_reads.fasta 
> > stats for data/raw_data/pacbio_reads.fasta
> > sum = 48280339, n = 3862, ave = 12501.38, largest = 70567
> > N50 = 19571, n = 840
> > N60 = 16667, n = 1107
> > N70 = 13410, n = 1427
> > N80 = 10361, n = 1833
> > N90 = 6819, n = 2399
> > N100 = 74, n = 3862
> > N_count = 0
> > Gaps = 0
> > ~~~
> > {: .bash}
> {: .solution}
> The Nanopore reads are in this file: data/raw_data/nanopore_reads.fastq. Also check the statistics of the Nanopore data.
> > ## Solution
> > ~~~
> > gt-workshop@gtworkshop-VirtualBox:~$ ./tools/assembly-stats-master/build/assembly-stats data/raw_data/nanopore_reads.fastq 
> > stats for data/raw_data/nanopore_reads.fastq
> > sum = 20742020, n = 1449, ave = 14314.71, largest = 73689
> > N50 = 22393, n = 324
> > N60 = 19386, n = 424
> > N70 = 15673, n = 544
> > N80 = 12226, n = 694
> > N90 = 7987, n = 900
> > N100 = 384, n = 1449
> > N_count = 0
> > Gaps = 0
> > ~~~
> > {: .bash}
> {: .solution}
> And finally the Illumina reads: data/raw_data/illumina_R1.fastq and data/raw_data/illumina_R2.fastq 
> > ## Solution
> > ~~~
> > ./tools/assembly-stats-master/build/assembly-stats data/raw_data/illumina_R1.fastq data/raw_data/illumina_R2.fastq 
> > stats for data/raw_data/illumina_R1.fastq
> > sum = 21846250, n = 87385, ave = 250.00, largest = 250
> > N50 = 250, n = 43693
> > N60 = 250, n = 52431
> > N70 = 250, n = 61170
> > N80 = 250, n = 69908
> > N90 = 250, n = 78647
> > N100 = 250, n = 87385
> > N_count = 0
> > Gaps = 0
> > -------------------------------------------------------------------------------
> > stats for data/raw_data/illumina_R2.fastq
> > sum = 21846250, n = 87385, ave = 250.00, largest = 250
> > N50 = 250, n = 43693
> > N60 = 250, n = 52431
> > N70 = 250, n = 61170
> > N80 = 250, n = 69908
> > N90 = 250, n = 78647
> > N100 = 250, n = 87385
> > N_count = 0
> > Gaps = 0
> > ~~~
> > {: .bash}
> {: .solution}
> Discuss the difference between the three platforms.
{: .challenge}

