---
title: "QC and evaluation of long read data"
teaching: 30
exercises: 60-90
questions:
- "How to determine basic statistics on read sets"
- "What do these statistics mean: are they good or bad?"
- "What are read technology specific errors?"  
objectives:
- "Learn how to get and interpret read set statistics"
- "Evaluate read technology specific errors based on read mappings"
keypoints:
- "PacBio and Nanopore have read lengths of many kilobases"
- "In practice quality of input DNA, library prep. and many other factors will determine read length distribution"
- "PacBio and Nanopore have different types of errors"
apps:
- ./tools/assembly-stats-master/build/assembly-stats 
---

To run the quality control on the read sets we use assembly-stats:
~~~
gt-workshop@gtworkshop-VirtualBox:~$ ./tools/assembly-stats-master/build/assembly-stats 
usage: stats [options] <list of fasta/q files>

Reports sequence length statistics from fasta and/or fastq files

options:
-l <int>
	Minimum length cutoff for each sequence.
	Sequences shorter than the cutoff will be ignored [1]
-s
	Print 'grep friendly' output
-t
	Print tab-delimited output
-u
	Print tab-delimited output with no header line
-v
	Print version and exit
~~~
{: .bash}

> ## Read statistics on PacBio
> The PacBio reads are in this file: data/pacbio_reads.subsample.fasta
> Get the read statistics on this data set and try to interpet the results
> > ./tools/assembly-stats-master/build/assembly-stats data/pacbio_reads.subsample.fasta 
> > stats for data/pacbio_reads.subsample.fasta
> > sum = 24433001, n = 1934, ave = 12633.40, largest = 70567
> > N50 = 19813, n = 424
> > N60 = 16833, n = 558
> > N70 = 13419, n = 719
> > N80 = 10444, n = 923
> > N90 = 6911, n = 1208
> > N100 = 74, n = 1934
> > N_count = 0
> > Gaps = 0
> {: .solution}
{: .challenge}

