---
title: "Comparison and visualization of long read assemblies"
teaching: 30
exercises: 120
questions:
- "How do you compare draft assemblies with mummerplots?"
- "How do you visualize repeat content?" 
objectives:
- "Knowing how to use mummerplots and how to interpet the results"
keypoints:
- "(Near-) exact contigs show up as a diagonal line in a mummerplot"
- "Repeat content shows up as dots / lines across the image"
apps:
- "nucmer"
- "mummerplot"
---

A common question to ask after finishing a de novo assembly is how does my new draft sequence looks like compared to either a reference genome or a previously created assembly.

To answer this question, we will use **mummerplot** to align two sequence files to each other and create a nice plot based on these alignments.

## Comparison to cultivar reference genome

## Comparison to TAIR10 reference genome

## Repeats in the assemblies

## Comparison of assemblies

## Mapping reads to the assemblies